ext/zip/php_zip.c

```diff
@@ -348,25 +349,62 @@ static int php_zip_parse_options(HashTable *options, zip_options *opts)
 #endif
 
 	if ((option = zend_hash_str_find(options, "remove_all_path", sizeof("remove_all_path") - 1)) != NULL) {
+		if (Z_TYPE_P(option) != IS_FALSE && Z_TYPE_P(option) != IS_TRUE) {
+			if (ZEND_CALL_USES_STRICT_TYPES(EG(current_execute_data))) {
```

Comment:
Is there any precedent for this? I don't remember us making strict_types distinctions for something other than parameters anywhere.